nano/node/blockprocessor.cpp

```diff
@@ -181,28 +208,57 @@ bool nano::block_processor::have_blocks ()
 	return have_blocks_ready ();
 }
 
-void nano::block_processor::add_impl (std::shared_ptr<nano::block> block)
+void nano::block_processor::add_impl (std::shared_ptr<nano::block> block, context ctx)
 {
+	release_assert (ctx.source != nano::block_source::forced);
 	{
 		nano::lock_guard<nano::mutex> guard{ mutex };
-		blocks.emplace_back (block);
+		blocks.emplace_back (entry{ block, std::move (ctx) });
 	}
 	condition.notify_all ();
 }
 
-auto nano::block_processor::process_batch (nano::unique_lock<nano::mutex> & lock_a) -> std::deque<processed_t>
+auto nano::block_processor::next () -> entry
+{
+	debug_assert (!mutex.try_lock ());
+	debug_assert (!blocks.empty () || !forced.empty ()); // This should be checked before calling next
+
+	if (!blocks.empty ())
```

Comment:
Is this correct? Previously, forced had priority. Now it doesn't.

Reply:
Fixed